### PR TITLE
feat: Add support for new file structure within Sipi folders and move Sipi tmp directory cleanup to separate route (DEV-1715)

### PR DESCRIPTION
**Pull request by irinaschubert**
### Task Description/Number

Issue Number: DEV-1715

This PR covers more than was initially intended.
- a new route `clean_temp_dir` was added
- a new folder structure for Sipi's file handling was introduced
- the most important changed files are: `sipi.init.lua`, `store.lua`, `upload.lua`, `upload_without_processing.lua` - the rest is just clean-up and adjustments to the new structure

### Basic Requirements

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

### PR Type

What kind of change does this PR introduce?

- [ ] fix: represents bug fixes
- [ ] refactor: represents production code refactoring
- [x] feat: represents a new feature
- [ ] docs: documentation changes (no production code change)
- [ ] chore: maintenance tasks (no production code change)
- [ ] test: all about tests: adding, refactoring tests (no production code change)
- [ ] other... Please describe:

### Does this PR introduce a breaking change?

- [x] Yes
- [ ] No
- [ ] Maybe (not 100% sure => check with FE)

### Does this PR change client-test-data?

- [ ] Yes (don't forget to update the JS-LIB team about the change)
- [x] No

### CODE REVIEW

**Review by BalduinLandolt — APPROVED** [review]
Looks good, I'm no Lua expert, though. I assume you tested it? :)

out of curiosity:  
1. did you test if SVA still has the desired behaviour
2. does "upload without processing" have permissions implications/requirements?

File: sipi/scripts/sipi.init.lua [hunk 1]
```diff
@@ -5,145 +5,225 @@ require "get_knora_session"
...
+
+    local filepath = get_tmp_filepath(prefix, identifier)
+    log("pre_flight - filepath: " .. filepath, server.loglevel.LOG_DEBUG)
+
+    if prefix == "tmp" then
+        -- always allow access to tmp folder
+        return 'allow', filepath
+    end
+
+    -- handle old way of file path - TODO: remove this block of code as soon as migration is done!
```
Comment (mpro7) [inline sipi/scripts/sipi.init.lua:178]: Maybe add the info what migration do you mean.

**Review by mpro7 — APPROVED** [review]
Looks OK, although I didn't look in detail on LUA files...

**Comment by irinaschubert** [comment]
> Looks good, I'm no Lua expert, though. I assume you tested it? :)
> 
> out of curiosity:
> 
> 1. did you test if SVA still has the desired behaviour
> 2. does "upload without processing" have permissions implications/requirements?

@BalduinLandolt Yes, I have tested it. We will see if it also works on DEV.
Regarding your questions:
1. Yes, I tested that, too.
2. The same permissions are required as with the normal upload route.

**Review by seakayone — APPROVED** [review]

**Pull request merged**